secret-provider/src/main/scala/io/lenses/connect/secrets/cache/TtlCache.scala

```diff
@@ -0,0 +1,45 @@
+package io.lenses.connect.secrets.cache
+
+import com.typesafe.scalalogging.LazyLogging
+
+import java.time.Clock
+import scala.collection.concurrent.TrieMap
+
+class TtlCache[V](implicit clock: Clock) extends LazyLogging {
+
+  private val cache = TrieMap[String, ValueWithTtl[V]]()
+
+  private def put(
+    key:   String,
+    value: ValueWithTtl[V],
+  ): ValueWithTtl[V] = {
+    cache.put(key, value)
+    value
+  }
+
+  private def get(
+    key:          String,
+    fnConditions: Seq[ValueWithTtl[V] => Boolean] = Seq(_ => true),
+  ): Option[ValueWithTtl[V]] =
+    cache.get(key).filter(value => fnConditions.forall(_(value)))
+
+  def cachingWithTtl(
+    key:                String,
+    fnCondition:        V => Boolean                       = _ => true,
+    fnGetMissingValue:  () => Either[Throwable, ValueWithTtl[V]],
```

Comment:
Is this a default value or throw? If so, maybe it's best to return an Option rather than Either. The caller would know what to do if None is present.